### PR TITLE
update doc for Load

### PR DESCRIPTION
**Pull request by nswamy**
addressed comments

### CODE REVIEW

File: python/mxnet/initializer.py [hunk 1]
```diff
@@ -209,6 +209,9 @@ def _init_default(self, name, _):
 class Load(object):
     """Initialize by loading data from file or dict.
```
Comment (madjam) [inline python/mxnet/initializer.py]: Initialize what?

Also change to 3rd person descriptive.

File: python/mxnet/initializer.py [hunk 2]
```diff
@@ -209,6 +209,9 @@ def _init_default(self, name, _):
...
 
+    **Note** Load will drop ``arg:`` or ``aux:`` from name and
+    initialize the variables that match with the prefix dropped.
+
     Parameters
     ----------
     param: str or dict of str->NDArray
```
Comment (madjam) [inline python/mxnet/initializer.py]: str -> `NDArray`?

File: python/mxnet/initializer.py [hunk 3]
```diff
@@ -217,6 +220,7 @@ class Load(object):
         Default initializer when name is not found in param.
```
Comment (madjam) [inline python/mxnet/initializer.py]: param in back ticks

File: python/mxnet/initializer.py [hunk 4]
```diff
@@ -217,6 +220,7 @@ class Load(object):
...
     verbose: bool
         Log source when initializing.
```
Comment (madjam) [inline python/mxnet/initializer.py]: Flag for enabling logging of source when initializing.

**Comment by piiswrong** [comment]
add an exmaple?

**Comment by madjam** [comment]
@piiswrong any idea why this failed?

**Comment by madjam** [comment]
I agree that an example would be helpful.

@nswamy  and I talked about the example. We felt there should be a how-to in our docs that discusses how to load/save models. We we create that we can add a link to that from here.

@piiswrong  is that ok with you?

**Comment by piiswrong** [comment]
ok.
note that is is mostly for partial loading. 
if you are loading the exact same model you can module.load directly.
this is useful when you want to initialize your new (with more params) model from a pretrained model.

**Pull request merged**